### ISSUE TITLE
Rats is half compatible with Fruit Trees

### ISSUE THREAD

**Issue by ReiDaTecnologia** [Krutoy242/Enigmatica2Expert-Extended#411]
## Description

Rats can bone meal them but can't collect them. Affect vanilla cocoa beans, all pam trees and Forestry's date and payaya (The other fruit trees from Forestry also don't work but as the fruits are in the leaves and they still drop fruits when chopped down, they aren't that bad).
**To Reproduce**

1. Get any of the tree mentioned above
2. Get a tamed rat to the tree (if you plant the tree's sapling near a tamed rat, it will instead break the sapling and if you grow the tree before he can do that, it will just stay/walk toward the previous location of the sapling until something else overwrite that previous command)
3. See the Rat not collecting the tree's fruit even if it is fully growth

**Modpack version** `v1.43.0`

**`debug.log` or Crash report**

This is the debug.log of a session of me testing all Pam trees, Forestry and cocoa beans. I had to do some pauses for several reasons so it explains the time skip of it.
[debug.zip](https://github.com/user-attachments/files/16885197/debug.zip)

**Comment by Krutoy242** [comment]
Thank you for the explanation. I will take a look what I can do.

**Comment by ReiDaTecnologia** [comment]
After checking Rats: RatN Edition and Pam's source code, i think it is because [Pam fruit blocks](https://github.com/MatrexsVigil/harvestcraft/blob/master/src/main/java/com/pam/harvestcraft/blocks/growables/BlockPamFruit.java) don't have "BlockCrops", "BlockBush" and blocks with the set Material "GOURD" which Rats is [looking for](https://github.com/Rebirth-of-the-Night/Rats-RatN-Edition/blob/master/src/main/java/com/github/alexthe666/rats/server/entity/ai/RatAIHarvestCrops.java).

**Comment by ReiDaTecnologia** [comment]
After months of work, i was able to improve the harvesting feature of rats mod. As i learned recently that [you can't just upload the modified jar to curseforge](https://github.com/Krutoy242/Enigmatica2Expert-Extended/issues/433#issuecomment-2476594846), people who want to have it, can just download from the fork's [release page](https://github.com/ReiDaTecnologia/Rats-RatN-Edition/releases/tag/Release). Any bug report and suggestions of improving the code or the harvesting feature is welcomed.

**Comment by Krutoy242** [comment]
Can you propose this changes to Rats:ROTN fork? Maybe they can just merge your PR? Your improvements looks very fitting.

**Comment by ReiDaTecnologia** [comment]
Sure but i prefer to test it in a real player environment (like in my world) first.

**Comment by Krutoy242** [comment]
Maybe, instead if making a separate mod, we could just make ZenScript mixin? This will make it easy to maintain and improve, since doesnt require things like dev environment, compilation and stuff like that.

**Comment by ReiDaTecnologia** [comment]
Well if you are willing to do that, i wish you good luck!

**Comment by Krutoy242** [comment]
Yes, I will try.
Only thing that buzzed me is by the first look on your code its kinda junky. Too much code repetitions, overload of static strings, unperformant calls, and so on.

**Comment by ReiDaTecnologia** [comment]
Yeah, i'm new to modding and not a god-like programmer. At least it works i guess...

**Comment by Krutoy242** [comment]
Seems like this feature will be implemented in future `Magiculture Integrations` mod, so I could close the thread.
